### PR TITLE
fix(76): Prevent the creation of 0-sized bitmaps

### LINKED ISSUE #76: crash with java.lang.IllegalArgumentException: width and height must be > 0
## Expected Behavior

If in some rare case when bitmap is not loaded code must exit gratefully or load an empty bitmap.

## Actual Behavior

Crashes the app. Observed only in customer crash reports.

## Steps to Reproduce the Problem

1. Check logs from customers.
2. See rare crash in logs;

## Specifications

- Version: ExtendedSplashScreen.Uno.WinUI Version="0.5.0-dev.60" 
- Platform: Android 14
- Observer on physical devices with frequencies:
```
Device              OS         Counts
-------------------------------------
Pixel 7 Pro         Android 14  10
Galaxy S21 Ultra 5G Android 14  1
Pixel 6             Android 14  1
```

## Stack Trace

```
Java.Interop.JniEnvironment.StaticMethods.CallStaticObjectMethod(JniObjectReference , JniMethodInfo , JniArgumentValue* )
Java.Interop.JniPeerMembers.JniStaticMethods.InvokeObjectMethod(String , JniArgumentValue* )
Android.Graphics.Bitmap.CreateBitmap(Int32 , Int32 , Config )
Nventive.ExtendedSplashScreen.ExtendedSplashScreen.GetBitmapFromView(View view)
Nventive.ExtendedSplashScreen.ExtendedSplashScreen.OnExitAnimation(Object sender, ExitAnimationEventArgs e)
AndroidX.Core.SplashScreen.SplashScreen.IOnExitAnimationListenerImplementor.OnSplashScreenExit(SplashScreenViewProvider )
AndroidX.Core.SplashScreen.SplashScreen.IOnExitAnimationListenerInvoker.n_OnSplashScreenExit_Landroidx_core_splashscreen_SplashScreenViewProvider_(IntPtr , IntPtr , IntPtr )
Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PPL_V(_JniMarshal_PPL_V , IntPtr , IntPtr , IntPtr )
java.lang.IllegalArgumentException: width and height must be > 0
android.graphics.Bitmap.createBitmap Bitmap.java:1197
android.graphics.Bitmap.createBitmap Bitmap.java:1163
android.graphics.Bitmap.createBitmap Bitmap.java:1111
android.graphics.Bitmap.createBitmap Bitmap.java:1070
mono.androidx.core.splashscreen.SplashScreen_OnExitAnimationListenerImplementor.n_onSplashScreenExit(Native Method)
mono.androidx.core.splashscreen.SplashScreen_OnExitAnimationListenerImplementor.onSplashScreenExit SplashScreen_OnExitAnimationListenerImplementor.java:31
androidx.core.splashscreen.SplashScreen$Impl31.setOnExitAnimationListener$lambda-0(SplashScreen.kt:459)
androidx.core.splashscreen.SplashScreen$Impl31.$r8$lambda$_AQ1xSddL7-kLq4WrGTBd4lTQaY(Unknown Source:0)
androidx.core.splashscreen.SplashScreen$Impl31$$ExternalSyntheticLambda0.onSplashScreenExit(Unknown Source:4)
android.window.SplashScreen$SplashScreenManagerGlobal.dispatchOnExitAnimation SplashScreen.java:271
android.window.SplashScreen$SplashScreenManagerGlobal.handOverSplashScreenView SplashScreen.java:258
android.app.ActivityThread.reportSplashscreenViewShown ActivityThread.java:4257
android.app.ActivityThread.lambda$syncTransferSplashscreenViewTransaction$1 ActivityThread.java:4274
android.app.ActivityThread.$r8$lambda$zFMoOisyD5hrqE37Jh7SYYMQGB8(Unknown Source:0)
android.app.ActivityThread$$ExternalSyntheticLambda5.run(Unknown Source:6)
android.view.Choreographer$CallbackRecord.run Choreographer.java:1339
android.view.Choreographer$CallbackRecord.run Choreographer.java:1348
android.view.Choreographer.doCallbacks Choreographer.java:952
android.view.Choreographer.doFrame Choreographer.java:878
android.view.Choreographer$FrameDisplayEventReceiver.run Choreographer.java:1322
android.os.Handler.handleCallback Handler.java:958
android.os.Handler.dispatchMessage Handler.java:99
android.os.Looper.loopOnce Looper.java:205
android.os.Looper.loop Looper.java:294
android.app.ActivityThread.main ActivityThread.java:8177
java.lang.reflect.Method.invoke(Native Method)
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run RuntimeInit.java:552
com.android.internal.os.ZygoteInit.main ZygoteInit.java:971
```

### PR DESCRIPTION
**Pull request by jeanplevesque**
GitHub Issue: Fixes #76 

## Proposed Changes

 - [x] Bug fix
 - [ ] Feature
 - [ ] Code style update (formatting)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build or CI related changes
 - [x] Documentation content changes
 - [ ] Other, please describe:

## What is the current behavior?

There is no validation on the bitmap dimensions when creating the copy of the native splash screen.

## What is the new behavior?

There is a validation on the bitmap dimensions to avoid creating a 0-sized bitmap.

## Impact on version

- [ ] **Major** (Public API was modified.)
  - Public constructs (class, struct, delegate, enum, etc.) were removed or renamed.
  - Public members were removed or renamed.
  - Public method signatures were changed or renamed.
- [ ] **Minor** (Public API was extended.)
  - Public constructs, members, or overloads were added.
- [x] **Patch** (Public API was unchanged.)
  - A bug in behavior was fixed.
  - Documentation was changed.
- [ ] **None** (The library is unchanged.)
  - Only code under the `build` folder was changed.
  - Only code under the `.github` folder was changed.

## Checklist

Please check that your PR fulfills the following requirements:

- [x] Documentation has been added/updated.
  - README.md was reviewed.
  - [BREAKING_CHANGES.md](../BREAKING_CHANGES.md) was updated (if you introduced a breaking change).
- [ ] Automated Unit / Integration tests for the changes have been added/updated.
- [x] Your conventional commits are aligned with the **Impact on version** section.

### CODE REVIEW

File: src/library/Shared/Implementations/ExtendedSplashScreen.Android.cs [hunk 1]
```diff
@@ -60,19 +61,29 @@ private static void OnExitAnimation(object sender, AndroidX.Core.SplashScreen.Sp
...
+			// It seems like the view can sometimes be invalid according to https://github.com/nventive/ExtendedSplashScreen/issues/76.
+			_isNativeViewValid = view.Width > 0 && view.Height > 0;
+
+			if (_isNativeViewValid.Value)
+			{
+				_splashBitmap = GetBitmapFromView(view);
+			}
+			else
+			{
+				Trace("Native view is invalid because at least one of its dimension is 0.");
```
Comment (takla21) [inline src/library/Shared/Implementations/ExtendedSplashScreen.Android.cs:75]: could we also log the view dimensions to have a bit more information?

File: src/library/Shared/Implementations/ExtendedSplashScreen.Android.cs [hunk 2]
```diff
@@ -60,19 +61,29 @@ private static void OnExitAnimation(object sender, AndroidX.Core.SplashScreen.Sp
 		{
 			Trace("Started OnExitAnimation.");
```
Comment (takla21) [inline src/library/Shared/Implementations/ExtendedSplashScreen.Android.cs:62]: Does this actually follow the appropriate log logic here?

**Review by takla21 — APPROVED** [review]

File: src/library/Shared/Implementations/ExtendedSplashScreen.Android.cs (hunk 2 above)
Comment (jeanplevesque) [reply src/library/Shared/Implementations/ExtendedSplashScreen.Android.cs:62]: All logs are traces in this class.

File: src/library/Shared/Implementations/ExtendedSplashScreen.Android.cs (hunk 1 above)
Comment (jeanplevesque) [reply src/library/Shared/Implementations/ExtendedSplashScreen.Android.cs:75]: I don't think it's relevant. The problem is that there's a 0 in there and that shouldn't be the case.

**Review by NerijusN — COMMENTED** [review]
```
_isNativeViewValid = view.Width > 0 && view.Height > 0;
if (_isNativeViewValid.Value)
```
makes things smooth. Let's release NuGet and see in practice.

**Pull request merged**